### DISCUSSION TITLE
Camera FOV Type

### DISCUSSION THREAD

**Discussion by ohyeah2389** [AIRLegend/aitrack#104] [Q&A] [answered]
I have a Logitech C310, which is listed on its official site https://www.logitech.com/en-us/products/webcams/c310-hd-webcam.960-000585.html as having a dFOV of 60. Is this correct for AITrack, or do I need to use vFOV or hFOV?
It doesn't seem to track my face if I hold it at greater than 30 degrees off angle, which seems very low.

**Comment by searching46dof** [comment, marked as answer]
You should use the diagonal FOV and not the vertical or horizontal FOV.
30 degrees seems to be the limit for detecting pitch while yaw seems to be a little more, possibly 45 degrees. If you are sending the UDP data to opentrack, it can map the 30degrees to 90degrees. I wouldn't recommend mapping more than 3x scaling factor since it creates more jitter than opentrack's filters can stabilize.

**Comment by AIRLegend** [comment]
Maybe the FOV parameter should be hidden as it doesn't seem to impact the position solver solution too much...

**Reply by searching46dof** [reply]
Actually the FOV does provide a more accurate face tracking. please review the proposed changes from https://github.com/searching46dof/aitrack/blob/master/AITracker/src/PositionSolver.cpp which simplify the initialization.
There were instances of transposed horizontal and vertical parameters which caused desensitization in yaw.

The changes would allow the FOV to be blank (=0.0) and would default to the horizontal and vertical resolution which provide the correct aspect ratio. When a FOV is supplied, it would also provide the correct scaling as well.

The FOV settings allow a a true 30degree yaw to be detected accurately as 30degree.
The FOV setting allows a more consistent head movement between different cameras.
e.g. if 2 cameras have the same resolution such as 640x480 with different actual FOV such as 56degree(telephoto) and 180degree(wide angle), they will show different horizontal pixel movement. Each will have the correct aspect ratio where 30degree pitch will show the same vertical pixel movement.  Without the FOV setting, you may need to turn your head more with the wide angle 180degree camera to be detected as the same 30degree.
